src/examples/algorithms/Readme.md

```diff
@@ -0,0 +1,32 @@
+# C++ NuPIC HelloWorld example
+
+This file serves two purposes: 
+
+## Demo C++ aplication 
+
+Using only C++ `nupic.core` implementations, currently the chain uses 
+` SpatialPooler > TemporalPooler` 
+
+More classes should be added as they are ported (Encoders, Classifier, Anomaly, ...)
+
+## Profiling code 
+
+For ongoing optimization goals, we need a simple but complete code to run benchmarks and profile. 
+You can easily change the constants (`EPOCHS, DIM,...`) and try this code on your branch. 
+
+### Using `valgrind` profiler (for memory, #calls usage)
```

Comment:
This is useful debug info. Perhaps @rhyolight can recommend a place for it that would be easier to find (although let's leave here for this PR - no need to change now).